cmd/performance-profile-creator/cmd/root.go

```diff
@@ -55,45 +55,90 @@ type ProfileData struct {
+			if err != nil {
+				return fmt.Errorf("failed to parse the cluster data: %v", err)
+			}
+
+			info, err := strconv.ParseBool(cmd.Flag("info").Value.String())
+			if err != nil {
+				return fmt.Errorf("failed to parse the info flag: %v", err)
+			}
+			if info {
+				showClusterData(cluster)
+				return nil
+			}
+
+			if !haveRequiredFlags(cmd, requiredFlags...) {
+				return fmt.Errorf("missing required flags: %s", strings.Join(requiredFlags, ", "))
+			}
+
+			profileCreatorArgsFromFlags, err := getDataFromFlags(cmd)
+			if err != nil {
+				return fmt.Errorf("failed to obtain data from flags %v", err)
+			}
+			profileData, err := getProfileData(profileCreatorArgsFromFlags, cluster)
+			if err != nil {
+				return err
+			}
+			createProfile(*profileData)
 			return nil
-		}
+		},
+	}
 
-		profileCreatorArgsFromFlags, err := getDataFromFlags(cmd)
-		if err != nil {
-			return fmt.Errorf("failed to obtain data from flags %v", err)
-		}
-		profileData, err := getProfileData(profileCreatorArgsFromFlags, cluster)
-		if err != nil {
-			return err
-		}
-		createProfile(*profileData)
-		return nil
-	},
+	root.PersistentFlags().IntVar(&pcArgs.ReservedCPUCount, "reserved-cpu-count", 0, "Number of reserved CPUs (required)")
+	root.PersistentFlags().BoolVar(&pcArgs.SplitReservedCPUsAcrossNUMA, "split-reserved-cpus-across-numa", false, "Split the Reserved CPUs across NUMA nodes")
+	root.PersistentFlags().StringVar(&pcArgs.MCPName, "mcp-name", "worker-cnf", "MCP name corresponding to the target machines (required)")
+	root.PersistentFlags().BoolVar(&pcArgs.DisableHT, "disable-ht", false, "Disable Hyperthreading")
+	root.PersistentFlags().BoolVar(&pcArgs.RTKernel, "rt-kernel", true, "Enable Real Time Kernel (required)")
+	root.PersistentFlags().BoolVar(pcArgs.UserLevelNetworking, "user-level-networking", false, "Run with User level Networking(DPDK) enabled")
+	root.PersistentFlags().StringVar(&pcArgs.PowerConsumptionMode, "power-consumption-mode", profilecreator.ValidPowerConsumptionModes[0], fmt.Sprintf("The power consumption mode.  [Valid values: %s]", strings.Join(profilecreator.ValidPowerConsumptionModes, ", ")))
+	root.PersistentFlags().StringVar(&pcArgs.MustGatherDirPath, "must-gather-dir-path", "must-gather", "Must gather directory path")
+	root.PersistentFlags().StringVar(&pcArgs.ProfileName, "profile-name", "performance", "Name of the performance profile to be created")
+	root.PersistentFlags().StringVar(&pcArgs.TMPolicy, "topology-manager-policy", kubeletconfig.RestrictedTopologyManagerPolicy, fmt.Sprintf("Kubelet Topology Manager Policy of the performance profile to be created. [Valid values: %s, %s, %s]", kubeletconfig.SingleNumaNodeTopologyManagerPolicy, kubeletconfig.BestEffortTopologyManagerPolicy, kubeletconfig.RestrictedTopologyManagerPolicy))
+	root.PersistentFlags().BoolVar(&pcArgs.Info, "info", false, "Show cluster information, disregard the other arguments")
+
+	requiredFlags = append(requiredFlags,
```

Comment:
Why haven't we initialized this at line 71 itself (at the time of their declaration) rather than appending it here?
```

var requiredFlags: []string{"reserved-cpu-count",
		"mcp-name",
		"rt-kernel",
		"must-gather-dir-path",
}
```

Reply:
I though about it and preferred the form I posted because this way the addition to the required set is closer to the flag definition. But the function is short enough that we can do this in one go like you suggested. Changed.